### PR TITLE
Api search book by filter

### PR DESCRIPTION
**Pull request by mvn-nale-dn**
## Description
Api search book by filter
## Reference Requirements
- Task ID: #108

## Reference Isuees
#{issue_id}

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-haphan** [comment]
LGTM

File: app/Http/Controllers/API/BookController.php [hunk 1]
```diff
@@ -21,6 +21,8 @@ public function index(Request $request)
         $books = new Book;
         if ($request->has('search')) {
             $books = $books->search($request->search);
+        } else {
```
Comment (at-vietpham) [inline app/Http/Controllers/API/BookController.php]: Không cần `else` ở đoạn này. Gộp chung trước `with` ở dưới là được.

**Review by ghost — CHANGES_REQUESTED** [review]

File: app/Traits/FilterTrait.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     * @param \Illuminate\Database\Eloquent\Builder $query  of Model.
+     * @param array                                 $fields fields
+     *
+     * @return void.
+     */
+    public function scopeFilter($query, $fields)
+    {
+        if (isset($fields)) {
+            foreach ($fields as $field => $value) {
+                if ($this->getColummnsFilter($field) !=null) {
```
Comment (at-vietpham) [inline app/Traits/FilterTrait.php]: Lưu `$this->getColummnsFilter($field)` vào 1 biến. tránh trường hợp gọi function này 2 lần.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by at-vietpham** [comment]
LGTM ⌨️

**Pull request merged**